### ISSUE TITLE
Transfer Learning with checkpoint - not able to use pre-trained weights when using distributed training with tf 1.14

### ISSUE THREAD

**Issue by purvang3** [tensorflow/models#9913]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/...

## 2. Describe the bug
machine having two gpus. 
system info: ubuntu 20
tf 1.14 docker

seems like it is creating two replicas but for second replica(replica_1), pretrained weights are not transferred as shown below.
using MirroredStrategy as described in steps to reproduce.
"""
2021-04-14 20:01:42.858990: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0, 1                                             
2021-04-14 20:01:42.859005: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:              
2021-04-14 20:01:42.859011: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 1                                                                     
2021-04-14 20:01:42.859015: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N N                                                                     
2021-04-14 20:01:42.859020: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 1:   N N                                                                     
2021-04-14 20:01:42.859070: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there m
ust be at least one NUMA node, so returning NUMA node zero                                                                                                        
2021-04-14 20:01:42.859374: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there m
ust be at least one NUMA node, so returning NUMA node zero                                                                                                        
2021-04-14 20:01:42.859744: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there m
ust be at least one NUMA node, so returning NUMA node zero                                                                                                        
2021-04-14 20:01:42.860059: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/device:GPU:0 with 7471 MB memory) -> physical GPU
 (device: 0, name: GeForce RTX 2080, pci bus id: 0000:02:00.0, compute capability: 7.5)                                                                           
2021-04-14 20:01:42.860094: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there m
ust be at least one NUMA node, so returning NUMA node zero                                                                                                        
2021-04-14 20:01:42.860370: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/device:GPU:1 with 7420 MB memory) -> physical GPU
 (device: 1, name: GeForce RTX 2070 SUPER, pci bus id: 0000:01:00.0, compute capability: 7.5)                                                                     
I0414 20:01:42.860693 140111631828800 cross_device_ops.py:1174] Device is available but not used by distribute strategy: /device:CPU:0                            
I0414 20:01:42.861099 140111631828800 cross_device_ops.py:1174] Device is available but not used by distribute strategy: /device:XLA_GPU:0                        
I0414 20:01:42.861286 140111631828800 cross_device_ops.py:1174] Device is available but not used by distribute strategy: /device:XLA_GPU:1                        
I0414 20:01:42.861404 140111631828800 cross_device_ops.py:1174] Device is available but not used by distribute strategy: /device:XLA_CPU:0                        
I0414 20:01:42.861697 140111631828800 run_config.py:558] Initializing RunConfig with distribution strategies.                                                     
I0414 20:01:42.861778 140111631828800 estimator_training.py:167] Not using Distribute Coordinator.

W0414 20:01:49.296050 140106681784064 variables_helper.py:141] Variable [BoxPredictor_0/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[273]], model variable shape: [[75]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296142 140106681784064 variables_helper.py:141] Variable [BoxPredictor_0/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 576, 273]], model variable shape: [[3, 3, 576, 75]]. This variable will not be initialized from the checkpo
int.
W0414 20:01:49.296213 140106681784064 variables_helper.py:141] Variable [BoxPredictor_1/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[546]], model variable shape: [[150]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296297 140106681784064 variables_helper.py:141] Variable [BoxPredictor_1/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 1024, 546]], model variable shape: [[3, 3, 1024, 150]]. This variable will not be initialized from the chec
kpoint.
W0414 20:01:49.296348 140106681784064 variables_helper.py:141] Variable [BoxPredictor_2/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[546]], model variable shape: [[150]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296404 140106681784064 variables_helper.py:141] Variable [BoxPredictor_2/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 512, 546]], model variable shape: [[3, 3, 512, 150]]. This variable will not be initialized from the checkp
oint.
W0414 20:01:49.296450 140106681784064 variables_helper.py:141] Variable [BoxPredictor_3/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[546]], model variable shape: [[150]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296488 140106681784064 variables_helper.py:141] Variable [BoxPredictor_3/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 256, 546]], model variable shape: [[3, 3, 256, 150]]. This variable will not be initialized from the checkp
oint.
W0414 20:01:49.296541 140106681784064 variables_helper.py:141] Variable [BoxPredictor_4/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[546]], model variable shape: [[150]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296593 140106681784064 variables_helper.py:141] Variable [BoxPredictor_4/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 256, 546]], model variable shape: [[3, 3, 256, 150]]. This variable will not be initialized from the checkp
oint.
W0414 20:01:49.296636 140106681784064 variables_helper.py:141] Variable [BoxPredictor_5/ClassPredictor/biases] is available in checkpoint, but has an incompatible
 shape with model variable. Checkpoint shape: [[546]], model variable shape: [[150]]. This variable will not be initialized from the checkpoint.
W0414 20:01:49.296684 140106681784064 variables_helper.py:141] Variable [BoxPredictor_5/ClassPredictor/weights] is available in checkpoint, but has an incompatibl
e shape with model variable. Checkpoint shape: [[3, 3, 128, 546]], model variable shape: [[3, 3, 128, 150]]. This variable will not be initialized from the checkp
oint.
W0414 20:01:49.297455 140106681784064 variables_helper.py:144] Variable [global_step] is not available in checkpoint
W0414 20:01:49.297571 140106681784064 deprecation_wrapper.py:119] From /root/main/python/object_detection/model_lib.py:282: The name tf.train.init_from_checkpoint
 is deprecated. Please use tf.compat.v1.train.init_from_checkpoint instead.

W0414 20:01:51.254761 140106690176768 variables_helper.py:144] Variable [BoxPredictor_0/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.254826 140106690176768 variables_helper.py:144] Variable [BoxPredictor_0/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.254869 140106690176768 variables_helper.py:144] Variable [BoxPredictor_0/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.254932 140106690176768 variables_helper.py:144] Variable [BoxPredictor_0/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255003 140106690176768 variables_helper.py:144] Variable [BoxPredictor_1/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255065 140106690176768 variables_helper.py:144] Variable [BoxPredictor_1/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255111 140106690176768 variables_helper.py:144] Variable [BoxPredictor_1/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255198 140106690176768 variables_helper.py:144] Variable [BoxPredictor_1/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255261 140106690176768 variables_helper.py:144] Variable [BoxPredictor_2/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255341 140106690176768 variables_helper.py:144] Variable [BoxPredictor_2/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255385 140106690176768 variables_helper.py:144] Variable [BoxPredictor_2/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255428 140106690176768 variables_helper.py:144] Variable [BoxPredictor_2/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255467 140106690176768 variables_helper.py:144] Variable [BoxPredictor_3/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255504 140106690176768 variables_helper.py:144] Variable [BoxPredictor_3/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255536 140106690176768 variables_helper.py:144] Variable [BoxPredictor_3/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255571 140106690176768 variables_helper.py:144] Variable [BoxPredictor_3/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255603 140106690176768 variables_helper.py:144] Variable [BoxPredictor_4/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255638 140106690176768 variables_helper.py:144] Variable [BoxPredictor_4/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255671 140106690176768 variables_helper.py:144] Variable [BoxPredictor_4/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255705 140106690176768 variables_helper.py:144] Variable [BoxPredictor_4/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255739 140106690176768 variables_helper.py:144] Variable [BoxPredictor_5/BoxEncodingPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255771 140106690176768 variables_helper.py:144] Variable [BoxPredictor_5/BoxEncodingPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255805 140106690176768 variables_helper.py:144] Variable [BoxPredictor_5/ClassPredictor/biases/replica_1] is not available in checkpoint
W0414 20:01:51.255837 140106690176768 variables_helper.py:144] Variable [BoxPredictor_5/ClassPredictor/weights/replica_1] is not available in checkpoint
W0414 20:01:51.255873 140106690176768 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/BatchNorm/beta/replica_1] is not available in 
checkpoint

W0414 20:44:39.683587 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/BatchNorm/gamma/replica_1] is not available in
 checkpoint
W0414 20:44:39.683625 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/BatchNorm/moving_mean/replica_1] is not availa
ble in checkpoint
W0414 20:44:39.683666 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/BatchNorm/moving_variance/replica_1] is not av
ailable in checkpoint
W0414 20:44:39.683706 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/depthwise_weights/replica_1] is not available 
in checkpoint
W0414 20:44:39.683747 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_1a_7x7/pointwise_weights/replica_1] is not available 
in checkpoint
W0414 20:44:39.683786 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2b_1x1/BatchNorm/beta/replica_1] is not available in 
checkpoint
W0414 20:44:39.683826 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2b_1x1/BatchNorm/gamma/replica_1] is not available in
 checkpoint
W0414 20:44:39.683867 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2b_1x1/BatchNorm/moving_mean/replica_1] is not availa
ble in checkpoint
W0414 20:44:39.683907 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2b_1x1/BatchNorm/moving_variance/replica_1] is not av
ailable in checkpoint
W0414 20:44:39.683947 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2b_1x1/weights/replica_1] is not available in checkpo
int
W0414 20:44:39.683989 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2c_3x3/BatchNorm/beta/replica_1] is not available in 
checkpoint
W0414 20:44:39.684027 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2c_3x3/BatchNorm/gamma/replica_1] is not available in
 checkpoint
W0414 20:44:39.684068 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2c_3x3/BatchNorm/moving_mean/replica_1] is not availa
ble in checkpoint
W0414 20:44:39.684108 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2c_3x3/BatchNorm/moving_variance/replica_1] is not av
ailable in checkpoint
W0414 20:44:39.684149 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Conv2d_2c_3x3/weights/replica_1] is not available in checkpo
int
W0414 20:44:39.684188 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Mixed_3b/Branch_0/Conv2d_0a_1x1/BatchNorm/beta/replica_1] is
 not available in checkpoint
W0414 20:44:39.684227 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Mixed_3b/Branch_0/Conv2d_0a_1x1/BatchNorm/gamma/replica_1] i
s not available in checkpoint
W0414 20:44:39.684267 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Mixed_3b/Branch_0/Conv2d_0a_1x1/BatchNorm/moving_mean/replic
a_1] is not available in checkpoint
W0414 20:44:39.684308 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Mixed_3b/Branch_0/Conv2d_0a_1x1/BatchNorm/moving_variance/re
plica_1] is not available in checkpoint
W0414 20:44:39.684347 140134943008512 variables_helper.py:144] Variable [FeatureExtractor/InceptionV2/Mixed_3b/Branch_0/Conv2d_0a_1x1/weights/replica_1] is not av
ailable in checkpoint.
"""

Also when I don't use distribute strategy, I can see same python process running on both gpus, but only one is being used checked with "nvidia-smi" as shown in images below (link)

https://github.com/purvang3/tf1.14_distributed_training/blob/main/Screen%20Shot%202021-04-15%20at%203.43.36%20PM.png

## 3. Steps to reproduce

added following lines in object_detection/model_main.py

distribution = tf.contrib.distribute.MirroredStrategy()
config = tf.estimator.RunConfig(model_dir=FLAGS.model_dir, train_distribute=distribution)

## 5. Additional context

Include any logs that would be helpful to diagnose the problem.

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device name if the issue happens on a mobile device:
- TensorFlow installed from (source or binary): tf1.14 docker container
- TensorFlow version (use command below):  1.14
- Python version: 3.6.8
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.0.130
- GPU model and memory: rtx 2070 super and 2080
